cwe_checker_rs/src/checkers/cwe_78/context/mod.rs

```diff
@@ -0,0 +1,563 @@
+
+        None
+    }
+
+    /// Creates a map from def terms to their corresponding pointer inference states
+    /// by taking the pointer inference state of the BlkStart node and updating it
+    /// for each def term in the block
+    pub fn create_pi_def_map(
+        &self,
+        block_start_node: NodeIndex,
+    ) -> Option<HashMap<Tid, PointerInferenceState>> {
+        if let Some(block_node) = self.get_pi_graph().node_weight(block_start_node) {
+            if let Some(pi_value) = self
+                .pointer_inference_results
+                .get_node_value(block_start_node)
+            {
+                let mut pi_def_map: HashMap<Tid, PointerInferenceState> = HashMap::new();
+
+                let pi_context = self.pointer_inference_results.get_context();
+                let mut new_pi_state = Some(pi_value.unwrap_value().clone());
+
+                for def in block_node.get_block().term.defs.iter() {
+                    // Add the pi state to the map that is available after the def was executed
+                    // If no state is available after the update_def() call, none is added
+                    new_pi_state = pi_context.update_def(&new_pi_state.unwrap(), def);
+                    if let Some(new_state) = new_pi_state.clone() {
+                        pi_def_map.insert(def.tid.clone(), new_state);
+                    }
+                }
+
+                return Some(pi_def_map);
+            }
+        } else {
+            panic!("Unexpected node index for BlkStart Node.");
+        }
+
+        None
+    }
+
+    /// Handles assignment and load definition updates
+    pub fn handle_assign_and_load(
+        &self,
+        state: State,
+        def: &Term<Def>,
+        var: &Variable,
+        input: &Expression,
+    ) -> State {
+        let mut new_state = state;
+        if let Some(taint) = new_state.get_register_taint(var) {
+            if let Taint::Tainted(_) = taint {
+                new_state.set_expression_taint_and_store_constants(
+                    &def.tid,
+                    var,
+                    input,
+                    &self.project.stack_pointer_register,
+                )
+            }
+        }
+
+        new_state
+    }
+
+    /// Handles store definition updates
+    pub fn handle_store(
+        &self,
+        state: State,
+        def: &Term<Def>,
+        address: &Expression,
+        value: &Expression,
+    ) -> State {
+        let mut new_state = state;
+        let target = match address {
+            Expression::Var(ram_target) => ram_target,
+            Expression::Subpiece { arg, .. } => {
+                let argument: &Expression = arg;
+                match argument {
+                    Expression::Var(ram_target) => ram_target,
+                    _ => panic!("Unexpected target format for Store instruction!"),
+                }
+            }
+            _ => panic!("Unexpected target format for Store instruction!"),
+        };
```

Comment:
I don't think we can rely on address expressions having this particular form. E.g. we may add a pass to remove unnecessary temporary registers, which could lead this code to panic in the process. So we should either try to handle generic address expressions, or at least prevent this code from panicking if the address expression is not as expected. Also: What about constant addresses, i.e. addresses of global variables?